src/kagglehub/hf_datasets.py

```
@@ -0,0 +1,62 @@
+        path: (string) Path to a file within the dataset
+        train_split_percent: (float) Optional split for train vs test data, defaults to 0.8
+        columns:
+            (list) Optional subset of columns to load from the file. Only used for CSV, TSV, Excel-like, feather,
+            and parquet files.
+        sheet_name:
+            (string, int, list, or None) Optional argument to be used for Excel-like files.
+            Defaults to 0 to select the first sheet. See pandas documentation for details:
+            https://pandas.pydata.org/docs/reference/api/pandas.read_excel.html
+        sql_query:
+            (string) Optional argument to be used for SQLite files. See pandas documentation for details:
+            https://pandas.pydata.org/docs/reference/api/pandas.read_sql_query.html
+    Returns:
+        A DatasetDict with test and train splits according to the provided train_split_percent
+    """
+    # These argument types produce a dictionary of DataFrames from pandas, but we need a single DataFrame
+    # to perform splits and loading into the DatasetDict. We validate upfront to respond to the user
+    # before doing any loading.
+    if sheet_name is None or isinstance(sheet_name, list):
+        raise ValueError(MULTIPLE_SHEETS_ERROR_MESSAGE)
+
+    result = load_pandas_dataset(handle, path, columns=columns, sheet_name=sheet_name, sql_query=sql_query)
+
+    # This shouldn't happen since we validated the sheet_name argument already, but we can check again to
+    # be robust and enable type safety in the else block.
+    if isinstance(result, dict):
+        raise ValueError(MULTIPLE_SHEETS_ERROR_MESSAGE)
+    else:
+        # Split the DataFrame into train/test splits based on the user-specified split
+        split_index = int(result.shape[0] * train_split_percent)
+        train_split = Dataset.from_pandas(result[:split_index])
+        test_split = Dataset.from_pandas(result[split_index:])
```

Shouldn't this default to random? For example, in the dataset used as an example in `load_hf_dataset.py`, entries are sorted in reverse order. Or use [`train_test_split`](https://huggingface.co/docs/datasets/main/en/package_reference/main_classes#datasets.Dataset.train_test_split) and somehow expose similar options, as I assume many people would want control over the splits.

---

I've updated the implementation pretty significantly. **tl;dr** We're just passing `kwargs` through to the underlying dependencies. So now you just get a `Dataset` and can do anything you want with it (like `train_test_split`, etc.). So users get the easy one-liner to load a HF Dataset from Kaggle, but then all the full-fledged features they want/enjoy from HF.